### ISSUE TITLE
Cert renewal via Gandi DNS plugin fails

### ISSUE THREAD

**Issue by erkston** [linuxserver/docker-swag#296]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior

Certbot should renew certificates automatically when using `VALIDATION=dns` and `DNSPLUGIN=gandi`

## Current Behavior
Certbot fails: `Renewal configuration file /etc/letsencrypt/renewal/DOMAIN.conf (cert: DOMAIN) produced an unexpected error: 'Namespace' object has no attribute 'certbot_plugin_gandi:dns_credentials'. Skipping.`

Further background: the Gandi DNS plugin was recently updated (#277) which comes with some configuration changes (https://github.com/obynio/certbot-plugin-gandi#faq). Any renewal configuration file created with the old plugin nomenclature will not renew and throw the error above.

Replacing 

```
authenticator = certbot-plugin-gandi:dns
certbot_plugin_gandi:dns_credentials = /config/dns-conf/gandi.ini
```

with

```
authenticator = dns-gandi
dns_gandi_credentials = /config/dns-conf/gandi.ini
```

in /etc/letsencrypt/renewal/DOMAIN.conf will allow certbot to renew without issue.

**Comment by nemchik** [comment]
As far as I can tell, we have never shipped `authenticator` as a line in `gandi.ini` based on the history https://github.com/linuxserver/docker-swag/commits/master/root/defaults/dns-conf/gandi.ini and in the previous repo https://github.com/linuxserver/docker-letsencrypt/commits/master/root/defaults/dns-conf/gandi.ini

We used to ship
```ini
certbot_plugin_gandi:dns_api_key=APIKEY
```
but that has changed to
```ini
dns_gandi_api_key=APIKEY
```
and should be automatically replaced in your `gandi.ini` when the container starts. The replacement happens with this line in the init:
https://github.com/linuxserver/docker-swag/blob/47cb0bffe315fe7030aa275be16ecaa27ec9065a/root/etc/cont-init.d/50-certbot#L41

I would advise you to only use `dns_gandi_api_key` in your `gandi.ini` file (all other lines we ship are comments and not required in the file).

p.s. for reference, python or certbot seem to treat dashes (`-`) and underscores (`_`) the same when reading these ini files, so `dns_gandi_api_key` and `dns-gandi-api-key` would both be read, however we have opted to stick with underscores.

**Comment by erkston** [comment]
dns-conf/gandi.ini is automatically replaced appropriately, but the older names were still present in my renewal conf file in /etc/letsencrypt/renewal/DOMAIN.conf

**Comment by nemchik** [comment]
That's a very important piece of information that I overlooked in your original post. Let me have a look at the best way to handle that.

**Comment by erkston** [comment]
> p.s. for reference, python or certbot seem to treat dashes (-) and underscores (_) the same when reading these ini files, so dns_gandi_api_key and dns-gandi-api-key would both be read, however we have opted to stick with underscores.

Note about this: I was actually still having trouble with renewal until I changed `dns-gandi-credentials` (dashes) to `dns_gandi_credentials` (underscores) in /etc/letsencrypt/renewal/DOMAIN.conf.

The original issue post had dashes but I have updated it to show underscores. Certbot will complain (`Input the path to your Gandi credentials INI file`) if dashes are used.
